webhook.go

```diff
@@ -231,7 +231,7 @@ func hookHandler(w http.ResponseWriter, r *http.Request) {
 
 		contentType := r.Header.Get("Content-Type")
 
-		if strings.Contains(contentType, "json") {
+		if strings.Contains(matchedHook.IncomingPayloadContentType, "json") || strings.Contains(contentType, "json") {
```

Comment:
This code won't actually behave the way you think it will...

Imagine this scenario for a moment: 
We have a malformed request incoming with form encoded data, but wrong header being sent from the server that the content is `json`. User, being smart, recognized the issue, and the vendor does not want to fix it, so the user turns to the webhook documentation, find that we support way to override the incoming payload content type, and happily sets the override of `IncomingPayloadContentType` to the `form` type.

Webhook gets to executing this line and checks for this condition, it asks itself, alright, does the override in the `IncomingPayloadContentType` contains `json`? - Nope. Then it proceeds further and ask itself, okay, I still have a chance to make this condition evaluate to true, does the content type header contain `json`? Yup! Got it, let me decode the payload as `json`, aaand... boom. Decoder explodes and user gets the very same cryptic error message you got that the payload could not be parsed. :-)

What you want to do is actually the following:

On the line L232, before we set the `contentType` variable to the header value, we should test whether the matched hook has incoming payload content type override. If it does, we set the `contentType` to the override and happily ignore the header value, otherwise, we set the `contentType` to the `r.Header.Get("Content-Type")`. Everything else remains the same.